Localization for datepicker and fullcalendar

GSLabIt opened the pull request:
- modified datepicker.min.js to fallback to en language
- created locale-all.js with all localization available up to now
- update build.json to load locale-all.js for datepicker.js

- added locale-all.js for fullcalendar
- fixed calendar.js to load localization based on user language

Commit:
Update datepicker.min.js

Commit:
Create locale-all.js

Added all available translations in a single file

Commit:
Update build.json

Loading datepicker translations from locale-all.js

Commit:
Merge pull request #1 from joezsweet/joez-localization

Update datepicker.min.js

Commit:
Update control.js

- set user language from boot info

Commit:
- added locale-all for fullcalendar

Commit:
- fixed calendar.js to load localization based on user language

netchampfaris commented:
@joezsweet 
datepicker.en.js handled translation using frappe's built in translation framework

joezsweet commented:
Yeah, but it doesn't handle first day of week,

Commit:
Update breadcrumbs.js

fix for error: 
`Traceback (most recent call last):
  File "/home/frappe/frappe-bench/apps/frappe/frappe/app.py", line 56, in application
    response = frappe.handler.handle()
  File "/home/frappe/frappe-bench/apps/frappe/frappe/handler.py", line 19, in handle
    execute_cmd(cmd)
  File "/home/frappe/frappe-bench/apps/frappe/frappe/handler.py", line 42, in execute_cmd
    ret = frappe.call(method, **frappe.form_dict)
  File "/home/frappe/frappe-bench/apps/frappe/frappe/__init__.py", line 903, in call
    return fn(*args, **newargs)
TypeError: get() takes exactly 1 argument (0 given)`

rised if `setup` breadcrumb is clicked and user has no permission

netchampfaris commented:
@joezsweet How about we handle translation using frappe.
For first day of the week, lets maintain a dict with `language` as key and `firstDay` as value.

joezsweet commented:
It sound to me not straightforward, I prefer to handle localization in a more standard way.

Can I ask what's the goal of maintaining translation in frappe and having workaround for first day?

Commit:
[minor] hide chart area when mandatory filters are not set

- using this.chart_area.hide() instead of this.chart_area.toggle(true);

Commit:
Merge pull request #2 from joezsweet/report-fix

[minor] hide chart area when mandatory filters are not set

Commit:
Merge pull request #3 from frappe/develop

Bringing up to date from frappe/frappe repo

Commit:
added en-GB localization

Commit:
Update locale-all.js

- removed en-GB
- added en-US
- changed first-day to 1 in en localization

Commit:
Modified date format for en and en-US localization

Commit:
Merge pull request #4 from frappe/develop

Merge from original fork (frappe/frappe)

netchampfaris commented on frappe/public/js/frappe/form/control.js:
```diff
@@ -625,7 +625,7 @@ frappe.ui.form.ControlDate = frappe.ui.form.ControlData.extend({
 		this.datepicker_options = {
-			language: "en",
+			language: frappe.boot.user.language || "en",
 			autoClose: true,
```
How about
```
		var lang = frappe.boot.user.language;
		if(!$.fn.datepicker.language[lang]) {
			lang = 'en'
		}

		this.datepicker_options = {
			language: lang,

```

joezsweet replied:
That's even better, i'll send a new commit

Commit:
Better handling for datepicker language fallback

GSLabIt reviewed (commented):
Sent new commit

joezsweet commented:
@netchampfaris @nabinhait  any chance to review code and include in version 8?

joezsweet commented:
Seems it is not a priority ..splitting it in different PR ...
